superset-frontend/src/views/CRUD/data/database/DatabaseModal.tsx

```diff
@@ -205,6 +203,14 @@ const DatabaseModal: FunctionComponent<DatabaseModalProps> = ({
 
       fetchResource(id)
         .then(() => {
+          // Altering 'extra' JSON before returning it to 'extra' text field
+          if (dbFetched && dbFetched.extra) {
+            const fetchedExtra = dbFetched?.extra;
+            const parsed = JSON.parse(fetchedExtra ?? defaultExtra);
+            delete parsed.metadata_cache_timeout;
+            delete parsed.schemas_allowed_for_csv_upload;
+            dbFetched.extra = JSON.stringify(parsed) ?? defaultExtra;
```

Comment:
I think you want `||` here in case the stringified parse is an empty string.

Reply:
No longer parsing JSON so I'm marking this as resolved.